### ISSUE TITLE
better document development workflow?

### ISSUE THREAD

**Issue by iantaylor-NOAA** [nmfs-stock-synthesis/stock-synthesis#143]
The contributing guide https://github.com/nmfs-stock-synthesis/stock-synthesis/blob/main/CONTRIBUTING.md is useful for those interested in modifying the code. However, it would be useful to have further notes on the development workflow beyond that point. 

In particular, I would find it useful to have written down a preferred set of steps for the SS team related to resolving an issue, such as (only with more detail):
- adding to a release milestone, 
- adding to a project,
- creating a pull request and requesting a review,
- merging a branch, 
- deleting the branch,
- updating the issue labels, 
- closing the issue, and 
- updating the project status.

I don't want to make things too complex, but it will be easier to have all this become routine if we can repeat a similar process each time. 

Now that this repository is public, we can create wiki pages at https://github.com/nmfs-stock-synthesis/stock-synthesis/wiki, so that might be a reasonable place to take note of this workflow and revise it as things evolve.

**Comment by k-doering-NOAA** [comment]
Great idea, Ian. It would be great to establish more norms around our workflow and have a written down reference guide. I think for anything that is just for the internal SS development team, we can have a wiki page, and anything that applies more broadly to contributors can go in the contributing guide? Does this division make sense?

Another part of this could be which checks we run and when. We now can more frequently build SS through github actions b/c we don't have limits on storage or action limits, and it is much easier to run jenkins now that it pulls the ss source code from github.

**Comment by k-doering-NOAA** [comment]
@Rick-Methot-NOAA and @iantaylor-NOAA , I put together a first version of the maintainer's wiki page: https://github.com/nmfs-stock-synthesis/stock-synthesis/wiki/Stock-Synthesis:-practices-for-maintainers . I tried to document our practices for working as I understand them.

Could you take a look and either make changes yourself or let me know what else should be added?

**Comment by iantaylor-NOAA** [comment]
Looks good to me. Thanks for putting this together.

One change I can think of is to add a step in the release checklist to remove the "resolved" label from any issues that were included in the release. And those issues should be closed at that time if they weren't already. There's more discussion of resolved issues at https://github.com/nmfs-stock-synthesis/stock-synthesis/issues/122 (maybe that issue can now be closed now that this wiki page is available).

Lastly, I see in #122 the tip to search for "is:issue is:open -label:resolved" to find open issues that need to be worked on. Would it make sense to add things like that to this page?

**Comment by k-doering-NOAA** [comment]
@iantaylor-NOAA great ideas, I think those should be added! Feel free to add yourself, or let me know if you want me to do it. Thanks for the input!

**Comment by iantaylor-NOAA** [comment]
I just made the two changes suggested in https://github.com/nmfs-stock-synthesis/stock-synthesis/issues/143#issuecomment-828666378. I also just learned that the history of changes for wiki pages on GitHub is accessed by clicking on the "[N] revisions" immediately below the title of the page.

My original hope in opening this issue has been more than satisfied by the new wiki page, but we should probably leave this issue open until @Rick-Methot-NOAA has time to look things over and suggest further changes.
